### ISSUE TITLE
Encoding issue when piping output to a file

### ISSUE THREAD

**Issue by gilesknap** [gilesknap/gphotos-sync#6]
See report from @pierrerigal below

The cause is described here https://stackoverflow.com/questions/5530708/can-i-redirect-unicode-output-from-the-console-directly-into-a-file

------------------------------

There is something weird, I can't redirect stdout. Each time I ran your script trying to redirect stdout to a log file, I fails with an error : 

`pi@raspberrypi:~/pyenvs/gphotos-sync/bin $ /home/pi/pyenvs/gphotos-sync/bin/gphotos-sync /media/toshiba/ &> /home/pi/pyenvs/gphotos-sync/var/gphotos.log
`
Then I have this in log file :  

```
pi@raspberrypi:/media/toshiba $ tail -f /home/pi/pyenvs/gphotos-sync/var/gphotos.log

Indexing Drive Folders ...
root_id = 0AJE60RrTQzeIUk9PVA
Resolving paths ...
Drive Folders scanned.


Indexing Drive Files ...

Done.

Saving Database ...
Database Saved.

Traceback (most recent call last):
  File "/home/pi/pyenvs/gphotos-sync/bin/gphotos-sync", line 7, in <module>
    g.main()
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/Main.py", line 168, in main
    self.start(args)
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/Main.py", line 137, in start
    self.drive_sync.index_drive_media()
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/GoogleDriveSync.py", line 214, in index_drive_media
    self.write_media(media, msg, False)
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/GoogleDriveSync.py", line 153, in write_media
    print(msg)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 40: ordinal not in range(128)

**Comment by gilesknap** [comment]
All output is now via logging and I have checked that I can redirect when logging album names with extended ASCII.

@pierrerigal please can you test for me when you have a moment? Thanks.

**Comment by pierrerigal** [comment]
Tested on master 

gphotos-sync --flush-index /media/toshiba/ &> var/gphotos.log

```
pi@raspberrypi:~/pyenvs/gphotos-sync $ tail -f var/gphotos.log

Indexing Drive Folders ...
root_id = 0AJE60RrTQzeIUk9PVA
Resolving paths ...
Drive Folders scanned.


Indexing Drive Files ...

Done.

Saving Database ...
Database Saved.

Traceback (most recent call last):
  File "/home/pi/pyenvs/gphotos-sync/bin/gphotos-sync", line 7, in <module>
    g.main()
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/Main.py", line 168, in main
    self.start(args)
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/Main.py", line 137, in start
    self.drive_sync.index_drive_media()
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/GoogleDriveSync.py", line 214, in index_drive_media
    self.write_media(media, msg, False)
  File "/home/pi/pyenvs/gphotos-sync/local/lib/python2.7/site-packages/gphotos/GoogleDriveSync.py", line 153, in write_media
    print(msg)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 40: ordinal not in range(128)
```

**Comment by gilesknap** [comment]
Strange, I reproduced the problem on my machine with an album that had extended ascii in its name and the fix worked in that case.
Also I note in your original post that the first file listed after indexing drive files did not seem to contain extended ascii.
Are you able to redirect a simple python script that just contains 'print "test" '?

**Comment by pierrerigal** [comment]
```
pi@raspberrypi:~ $ vi test.py
pi@raspberrypi:~ $ python test.py
test
pi@raspberrypi:~ $ python test.py > test.log
pi@raspberrypi:~ $ tail -f test.log
test
^C
```

No problem with this test...

**Comment by gilesknap** [comment]
Thanks for that.
In that case please could you try the following modification.
line 144 in Main.py
        formatter = logging.Formatter('%(name)s: %(message)s')
change to
        formatter = logging.Formatter(u'%(name)s: %(message)s')

(do setup build / install before trying)

**Comment by gilesknap** [comment]
If that still does not work please try adding this line directly after line 144:-
``logging._defaultFormatter = logging.Formatter(u"%(message)s")``

**Comment by pierrerigal** [comment]
Sorry, in which file ? 
Thx.

**Comment by gilesknap** [comment]
oops!
Main.py

**Comment by pierrerigal** [comment]
That's working !

I will let indexing end this night to see if it start downloading. 

(indexing is very slow on raspberry, it start going slow at 10K+ photos.

**Comment by gilesknap** [comment]
The speed problem might be memory related. Can you send the result of  
``top -o '%MEM'``
while gphotos-sync is running, please?

**Comment by pierrerigal** [comment]
![capture d ecran 2017-11-27 a 07 59 41](https://user-images.githubusercontent.com/895267/33254439-14802d74-d349-11e7-9026-0f304b714d98.png)

Still running after a night. Has indexed only 16K photos in about 8hours.

**Comment by pierrerigal** [comment]
Maybe I should build the database from my mac, and then let raspberry only do overnight updates.

**Comment by pierrerigal** [comment]
Actually it's 10s to index 1 single photo ^^ 

On more screenshot with htop, looks like it's not a memory problem since I still have free mem.

![capture d ecran 2017-11-27 a 09 55 00](https://user-images.githubusercontent.com/895267/33258246-28140120-d359-11e7-9a12-51228eefe2c0.png)

**Comment by pierrerigal** [comment]
One more idea : my raspberry is backing up file over LAN via wifi. So sql database is not on the raspberry. Maybe an option to keep sql db localy is a better idea (with a parameter --db-path).

**Comment by gilesknap** [comment]
performance issue moved to new ticket. Encoding issue fixed in v1.01
